Mathlib/Analysis/Calculus/Implicit.lean

```diff
@@ -440,4 +460,97 @@ theorem to_implicitFunction (hf : HasStrictFDerivAt f f' a) (hf' : range f' = 
 
 end FiniteDimensional
 
+section Bivariate
+
+/-!
+### Bivariate case
+
+Here we identify `E` with `X × Y`, `G` with `X` and `g : E → G` with the first projection out of
+`X × Y`. For consistency of notation `F` becomes `Z`. Now `f : X × Y → Z` is explicitly bivariate,
+and if its partial derivative with respect to `y` is invertible then the kernels of `f` and `g` are
+complementary. In such circumstances we may construct an instance of the `ImplicitFunctionData` data
+structure provided above and extract implicit function `ψ : X → Y` whose germ is unique. This
+functionality is wrapped by `HasStrictFDerivAt.implicitFunOfBivariate`.
+
+## TODO
+
+* Provide another version with curried `f : X → Y → Z` and with technical assumptions made on the
+  partial derivatives.
+* Establish results about higher derivatives.
+-/
+
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+variable {X : Type*} [NormedAddCommGroup X] [NormedSpace 𝕜 X] [CompleteSpace X]
+variable {Y : Type*} [NormedAddCommGroup Y] [NormedSpace 𝕜 Y] [CompleteSpace Y]
+variable {Z : Type*} [NormedAddCommGroup Z] [NormedSpace 𝕜 Z] [CompleteSpace Z]
+
+/-- Given linear map `fx : X →L[𝕜] Z`, linear equivalence `fy : Y ≃L[𝕜] Z` and that
+`HasStrictFDerivAt f (fx.coprod fy) p₀`, we construct an object of type `ImplicitFunctionData`
```

Comment:
In addition to what you already write, it might be enlightening for the user to emphasize that the true assumption here is that the partial derivative with respect to `y` is invertible, although it is a little bit hidden behind the formalism.